platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblemGroup.java

```diff
@@ -37,11 +38,21 @@ public DefaultProblemGroup(String groupId, String displayName) {
     }
 
     public DefaultProblemGroup(String name, String displayName, @Nullable ProblemGroup parent) {
-        this.name = name;
-        this.displayName = displayName;
+        validateFields(name, displayName);
+        this.name = TextUtil.replaceLineSeparatorsOf(name, "");
```

Comment:
💭 Would it be better to fail on line separators instead of silently removing them?  Seems like there isn't a need to supply strings with separators here.